src/common/escape.cc

```diff
@@ -112,6 +114,60 @@ void escape_xml_attr(const char *buf, char *out)
 	*o = '\0';
 }
 
+// applies hex formatting on construction, restores on destruction
+struct hex_formatter {
+  std::ostream& out;
+  const char fill;
+  const std::ostream::fmtflags flags;
```

Comment:
Perhaps mark that these are being saved rather than used.

Reply:
renamed to `old_name` and `old_flags`